packages/@aws-cdk/aws-ec2/test/test.vpc.ts

```diff
@@ -1067,6 +1067,56 @@ export = {
       test.deepEqual(subnetIds.length, 1);
       test.deepEqual(subnetIds[0], subnet.subnetId);
       test.done();
+    },
+
+    'subnet created from subnetId'(test: Test) {
+      // GIVEN
+      const stack = getTestStack();
+
+      // WHEN
+      const subnet = Subnet.fromSubnetId(stack, 'subnet1', 'pub-1');
+
+      // THEN
+      test.deepEqual(subnet.subnetId, 'pub-1');
+      test.done();
+    },
+
+    'Referencing AZ throws error when subnet created from subnetId'(test: Test) {
+      // GIVEN
+      const stack = getTestStack();
+
+      // WHEN
+      const subnet = Subnet.fromSubnetId(stack, 'subnet1', 'pub-1');
+
+      // THEN
+      test.throws(() => subnet.availabilityZone);
```

Comment:
If you are testing for an exception, please test the exception message as well (who's to say thisdoesn't throw "`undefined is not a function`" accidentally)

Reply:
verified error message as well.